WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalUpdateProgress.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+
+/// Modal presented when a firmware update is being installed
+///
+final class CardPresentModalUpdateProgress: CardPresentPaymentsModalViewModel {
+    /// Amount of update that has been completed
+    private let progress: Float
+
+    /// Called when cancel button is tapped
+    private let cancelAction: (() -> Void)?
+
+    let textMode: PaymentsModalTextMode = .fullInfo
+    let actionsMode: PaymentsModalActionsMode
+
+    var topTitle: String
+
+    var topSubtitle: String? = nil
+
+    let image: UIImage
+
+    let primaryButtonTitle: String? = nil
+
+    let secondaryButtonTitle: String? = Localization.cancel
+
+    let auxiliaryButtonTitle: String? = nil
+
+    let bottomTitle: String?
+
+    var bottomSubtitle: String? = Localization.message
```

Comment:
The bottomSubtitle prompting them to update and not cancel remains on the screen when the update is complete. Perhaps, like topTitle, it should change when the update is complete?

<img width="195" alt="Screen Shot 2021-09-21 at 1 56 24 PM" src="https://user-images.githubusercontent.com/1595739/134246686-1fe27c29-addd-4aaa-b3c1-882fc6449088.png">

Reply:
Yes, I agree with this thanks @allendav @koke

Reply:
Changed in 4b3e505ec